File: 24th_oct/Basic_coding_tasks/q2/solution.cpp
#include<iostream>
using namespace std;

enum Divisibility {
    EVEN,
    ODD
};
// 0 = even
// 1 = odd

Divisibility check_divisibility(int n){
    if (n%2 == 0){
        return Divisibility::EVEN;
    }
    else{
        return Divisibility::ODD;
    }

}

int main(){
    int x = 69934;
    Divisibility result = check_divisibility(x);
    if (result == Divisibility::EVEN){
        cout<< "even \n";
    } else{
        cout<< "odd \n";
    }
    
    //cout<< result<< "\n";
    return 0;
}

File: 24th_oct/Basic_coding_tasks/q2/solution.py
from enum import Enum

class Divisibility(Enum):
    EVEN = 0,
    ODD = 1

def check_divisibilty(x: int) -> Divisibility:
    if (x%2 == 0):
        return Divisibility.EVEN
    else:
        return Divisibility.ODD
    
result = check_divisibilty(8489437997493749734)
if (result == Divisibility.EVEN):
    print("even")
else:
    print("odd")
